tests/test_cases/test_array_buses/test_array_buses.py

```diff
@@ -0,0 +1,81 @@
+import cocotb
+from cocotb.clock import Clock
+from cocotb.drivers import BusDriver
+from cocotb.monitors import BusMonitor
+from cocotb.triggers import RisingEdge
+
+
+class TestDriver(BusDriver):
+    _signals = ["data", "valid"]
+
+    def __init__(self, entity, name, clock, **kwargs):
+        BusDriver.__init__(self, entity, name, clock, **kwargs)
+        self.bus.valid <= 0
+
+    @cocotb.coroutine
+    def _driver_send(self, transaction, sync=True):
+        clkedge = RisingEdge(self.clock)
+        if sync:
+            yield clkedge
+
+        self.log.info("Sending {}".format(transaction))
+        self.bus.valid <= 1
+        self.bus.data <= transaction
+
+        yield clkedge
+        self.bus.valid <= 0
```

Comment:
Please use the new syntax.
```suggestion
    async def _driver_send(self, transaction, sync=True):
        clkedge = RisingEdge(self.clock)
        if sync:
            await clkedge

        self.log.info("Sending {}".format(transaction))
        self.bus.valid <= 1
        self.bus.data <= transaction

        await clkedge
        self.bus.valid <= 0
```